Bottom: 4ea69795c21e3d750b6ef97c0512c601b8aff54f
Top:    0d07bfc760ad9d8c2b19559f5f23a55f1fc7e012
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:34:57 +0000

Try debugging draw speed




---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index fc7c352..b48c6bf 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2186,13 +2186,12 @@ Benchmark (int argc, char **argv, int x, int y)
     {
       gdk_window_invalidate_rect (gport->drawing_area->window, NULL, 1);
       gdk_window_process_updates (gport->drawing_area->window, FALSE);
-      gdk_display_sync (display);
       time (&end);
       i++;
     }
   while (end - start < 10);
 
-  printf ("%g redraws per second\n", i / 10.0);
+  printf ("%g redraws per second\n", (double)i / (double)(end-start));
 
   return 0;
 }
